lib/web_ui/lib/src/engine/surface/scene_builder.dart

```diff
@@ -97,6 +97,14 @@ class SurfaceSceneBuilder implements ui.SceneBuilder {
     if (matrix4.length != 16) {
       throw ArgumentError('"matrix4" must have 16 entries.');
     }
+    if (_surfaceStack.length == 1) {
+      // Top level transform contains view configuration to scale
+      // scene to devicepixelratio. Use identity instead since CSS uses
+      // logical device pixels.
+      assert(matrix4[0] == EngineWindow.browserDevicePixelRatio &&
+          matrix4[5] == EngineWindow.browserDevicePixelRatio);
```

Comment:
Removed the check for test environment since there are many SceneBuilder raw tests outside the context of an app that pushes root transform for now (we really need a ui.Window level debugDevicePixelRatio, i noticed some hacked in TestWindow.devicePixelRatioTestValue etc in the code base used for viewport_test).